api/src/main/java/org/openmrs/api/context/Context.java

```diff
@@ -1189,11 +1189,26 @@ private static void checkForDatabaseUpdates(Properties props) throws DatabaseUpd
 	 *
 	 * @param userInput (can be null) responses from the user about needed input
 	 * @throws DatabaseUpdateException if an error occurred while updating
-	 * @throws InputRequiredException if user input is required
 	 * @since 1.5
+	 * @deprecated as of 2.4; see {@link #updateDatabase()}
+	 * 
+	 */
+	@Deprecated
+	public static void updateDatabase(Map<String, Object> userInput) throws DatabaseUpdateException {
+		DatabaseUpdater.executeChangelog();
+	}
+
+	/**
+	 * Updates the openmrs database to the latest. This is only needed if using the API alone. <br>
+	 * <br>
+	 * The typical use-case would be: Try to {@link #startup(String, String, String, Properties)},
+	 * if that fails, call this method to get the database up to speed.
+	 * 
+	 * @throws DatabaseUpdateException if an error occurred while updating
+	 * @since 2.4
 	 */
-	public static void updateDatabase(Map<String, Object> userInput) throws DatabaseUpdateException, InputRequiredException {
-		DatabaseUpdater.executeChangelog(null, userInput);
+	public static void updateDatabase() throws DatabaseUpdateException {
+		DatabaseUpdater.executeChangelog();
 	}
 
```

Comment:
To be honest, I think this existed to run the database updates if either the initialization filter or update filter didn't get called. I really can't find any example of this actually being used anywhere. Good point on the illegal argument exception. I was mostly concerned with eliminating the endless recursion [here](https://github.com/openmrs/openmrs-core/pull/3619/files#diff-3f2d7ddb866d3834bf8e662b8c79a69a8e85808c2ddbf17c9493310d9d86c5a2L145). Maybe the thing to do is just remove these methods and see what breaks?

Reply:
`public static void updateDatabase(Map<String, Object> userInput)` existed before and is now deprecated which makes sense.

`public static void updateDatabase()` is new since 2.4 and I was wondering why it is added. 

Definitely a +1 on getting rid of the the strange recursion.

Reply:
>> Maybe the thing to do is just remove these methods and see what breaks?

Yes, but breaking it "gently" by throwing exceptions but still allowing any code using this method to compile.

Reply:
Well, I missed the fact that this code would throw an exception, so the new method was really just intended to replace `public static void updateDatabase(Map<String, Object> userInput)` with a version with the arguments we actually use (none). Obviously, that's also a bad idea!